### ISSUE TITLE
Freezing at "Using /home/user/.cache/torch_extensions as PyTorch extensions root..."

### ISSUE THREAD

**Issue by mallorbc** [dredwardhyde/gpt-neo-fine-tuning-example#6]
Prreviously I was able to run the model, but getting loss overflows for any custom data or running out of RAM, but it seemed to work for the Netflix dataset.  I made a new conda environment and now I get a point where some Vram is allocated before it just freezes at this point.

python gpt_neo_xl_deepspeed.py 
Special tokens have been added in the vocabulary, make sure the associated word embedding are fine-tuned or trained.
Max length: 384
[2021-06-08 10:46:33,763] [INFO] [distributed.py:47:init_distributed] Initializing torch distributed with backend: nccl
[2021-06-08 10:46:33,899] [INFO] [logging.py:60:log_dist] [Rank 0] DeepSpeed info: version=0.3.13, git-hash=unknown, git-branch=unknown
[2021-06-08 10:46:34,019] [INFO] [engine.py:77:_initialize_parameter_parallel_groups] data_parallel_size: 1, parameter_parallel_size: 1
Using /home/user/.cache/torch_extensions as PyTorch extensions root...

I feel like this may be due to environment issues, perhaps cudatoolkit since that is not specified.  What cudatoolkit was used?  Does anyone have any idea what the issue may be?  See below for my env:

name: gpt_neo_train
channels:
  - nvidia
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - ca-certificates=2021.5.25=h06a4308_1
  - certifi=2021.5.30=py37h06a4308_0
  - cudatoolkit=11.1.74=h6bb024c_0
  - ld_impl_linux-64=2.33.1=h53a641e_7
  - libffi=3.3=he6710b0_2
  - libgcc-ng=9.1.0=hdf63c60_0
  - libstdcxx-ng=9.1.0=hdf63c60_0
  - ncurses=6.2=he6710b0_1
  - openssl=1.1.1k=h27cfd23_0
  - pip=21.1.1=py37h06a4308_0
  - python=3.7.10=hdb3f193_0
  - readline=8.1=h27cfd23_0
  - setuptools=52.0.0=py37h06a4308_0
  - sqlite=3.35.4=hdfb4753_0
  - tk=8.6.10=hbc83047_0
  - wheel=0.36.2=pyhd3eb1b0_0
  - xz=5.2.5=h7b6447c_0
  - zlib=1.2.11=h7b6447c_3
  - pip:
    - chardet==4.0.0
    - click==8.0.1
    - deepspeed==0.3.13
    - filelock==3.0.12
    - idna==2.10
    - importlib-metadata==4.5.0
    - joblib==1.0.1
    - ninja==1.10.0.post2
    - numpy==1.17.3
    - packaging==20.9
    - pandas==1.2.2
    - pillow==8.2.0
    - protobuf==3.17.2
    - psutil==5.8.0
    - pyparsing==2.4.7
    - python-dateutil==2.8.1
    - pytz==2021.1
    - regex==2021.4.4
    - requests==2.25.1
    - sacremoses==0.0.45
    - six==1.16.0
    - tensorboardx==1.8
    - tokenizers==0.10.3
    - torch==1.8.1+cu111
    - torchsummary==1.5.1
    - torchvision==0.9.1+cu111
    - tqdm==4.61.0
    - transformers==4.5.0
    - typing-extensions==3.10.0.0
    - urllib3==1.26.5
    - zipp==3.4.1
prefix: /home/user/anaconda3/envs/gpt_neo_train

**Comment by mallorbc** [comment]
I solved this issue.  The issue was that some of the required elements for DeepSpeed were not installed using pip.  You can check this by running ds_report.  I fixed this by cloning the Deepspeed repo tag 0.3.13 and building from the source.

Steps:
1. Clone the repo with the proper tag
2. cd in the repo
3. Run DS_BUILD_OPS=1 pip install .

This should install the proper version of DeepSpeed with all the needed dependencies.
